plugins/woocommerce/src/Blocks/BlockTypesController.php

```diff
@@ -155,6 +156,16 @@ public function register_blocks() {
 		}
 	}
 
+	public function register_block_metadata() {
+		$meta_file_path = WC_ABSPATH . 'assets/client/blocks/blocks-json.php';
+		if ( function_exists( 'wp_register_block_metadata_collection' ) && file_exists( $meta_file_path ) ) {
+			wp_register_block_metadata_collection(
+				WC_ABSPATH . 'assets/client/blocks/',
```

Comment:
Using `WC_ABSPATH` does seem like the right approach here, but seems to break in my local environment as I have WooCommerce symlinked. I left a question in core around this: https://wordpress.slack.com/archives/C02QB2JS7/p1729539333146749

Reply:
This may not be a blocker though, given it would just default to the old way if things are symlinked.

Reply:
Realizing this last comment is a bit ambiguous.
I mean that the above issue isn't a blocker and this PR could be merged when the other items are addressed. We can create a follow up PR around this issue if changes on our end are necessary.

Reply:
I see - so it could be failing because it's checking `/Users/dev/projects/woocommerce/assets/client/blocks` against `/srv/http/wordpress/wp-content/plugins` and thinks it's not a specific plugin? That's unfortunate, because some hosts may be linking plugins in production as well.

Reply:
Yeah exactly, and it sounds like from this issue and conversation that they won't change it: https://core.trac.wordpress.org/ticket/62140

We can update it on our end, and instead of relying on `WC_ABSPATH` heavily we actually make use of the plugin directory path, as this should still work.

Reply:
Several managed hosts run Woo symlinked and managed from another path, would this be a problem?

Reply:
`$this->asset_api->package->get_path( 'assets/client/blocks/blocks-json.php' );` Should work in that file, otherwise to make it shorter you can inject the package in bootstrap and use it directly.

Reply:
## Symlinking - What happens?

>Several managed hosts run Woo symlinked and managed from another path, would this be a problem?

It means that they won't be able to take advantage of the optimization, but it won't crash or anything. If the paths don't match up, it simply falls back to the old behavior.

## Trying `package->get_path()`

>`$this->asset_api->package->get_path( 'assets/client/blocks/blocks-json.php' );` Should work in that file

This is a fatal error: `Fatal error: Uncaught Error: Cannot access private property Automattic\WooCommerce\Blocks\Assets\Api::$package`.

I was able to find ` $meta_file_path3 = \Automattic\WooCommerce\Blocks\Package::container()->get( \Automattic\WooCommerce\Blocks\Domain\Package::class )->get_path( 'assets/client/blocks/blocks-json.php' );`, but as far as I can tell, this is the exact same as `WC_ABSPATH . 'assets/client/blocks/blocks-json.php';`. For example, when I have `plugins/woocommerce` linked to `/tmp2/woocommerce`, both are `/tmp2/woocommerce/assets/client/blocks/blocks-json.php`.

From trying to pull apart the DI, Package seems to be initialized with the abs path:
```
/tmp2/woocommerce/src/Blocks/Package.php:94 - __construct("11.8.0-dev", "\/tmp2\/woocommerce", {})
```

It might be coming from this `__DIR__` in `src/Blocks/Package.php`:
```
	public static function container( $reset = false ) {
		static $container;
		if (
			! $container instanceof Container
			|| $reset
		) {
			$container = new Container();
			// register Package.
			$container->register(
				NewPackage::class,
				function ( $container ) {
					// leave for automated version bumping.
					$version = '11.8.0-dev';
					return new NewPackage(
						$version,
						dirname( __DIR__, 2 ),
						new FeatureGating()
					);
				}
			);
```

## Changing `WC_ABSPATH` -> `WP_PLUGIN_DIR`

Changing the manifest registration is easy enough:
```
	public function register_block_metadata() {
		$meta_file_path = WP_PLUGIN_DIR . '/woocommerce/assets/client/blocks/blocks-json.php';
		if ( function_exists( 'wp_register_block_metadata_collection' ) && file_exists( $meta_file_path ) ) {
			wp_register_block_metadata_collection(
				WP_PLUGIN_DIR . '/woocommerce/assets/client/blocks/',
				$meta_file_path
			);
		}
	}
```

However, the block registrations also need to be changed to match:
```
	protected function register_block_type() {
        // ..truncated..
		$metadata_path = $this->asset_api->get_block_metadata_path( $this->block_name );
```

This I don't know how to do safely because there are so many layers of OO and DI. The Api.php has:
```
	public function get_block_metadata_path( $block_name, $path = '' ) {
		$path_to_metadata_from_plugin_root = $this->package->get_path( 'assets/client/blocks/' . $path . $block_name . '/block.json' );
```

I think that depends on how the Package class gets instantiated, but I'd rather leave that up to someone else more familiar with the codebase :pray:.

Reply:
I added a bypass for the warning message here: 4a0f4c11b3a4986318bea2df17e399851c040303
This is the same approach we used in JP after the notices were causing alarm with some customers.
I just went through a clean install by building a woocommerce zip, setting WP_DEBUG to be true, and installing woocommerce off a symlinked directory to make sure the bypass was working.

I have also added a comment to https://core.trac.wordpress.org/ticket/62140 but it doesn't seem to be getting much traction.